File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */

/*
    [5,8,50,50,8,4,7,5]
     0 1 2 3 4 5 6 7   

*/
var maxArea = function(height) {
    let left = 0, right = height.length - 1, max = 0, currMax;
    
    while (left < right) {
        currMax = (right - left) * Math.min(height[left], height[right]);
        height[left] < height[right] ? left++ : right--;
        max = Math.max(max, currMax);
    }
    return max;
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        max_area = 0
        curr_area = 0
    
        while left < right:
            curr_area = (right - left) * min(height[left], height[right])
            if height[left] <= height[right]:
                left += 1 
            else: 
                right -= 1
            max_area = max(max_area, curr_area)
            
        return max_area